cmake/options.cmake

```diff
@@ -233,6 +233,11 @@ macro(drake_setup_options)
     "Google command-line flags processing library")
 
   drake_system_dependency(
```

Comment:
FYI: This implies that pybind11 is required (unless Python is disabled), but you can choose to use the system version. I'm happy to adjust if folks aren't happy with that arrangement.